apps/agreement/contracts/Agreement.sol

```diff
@@ -110,238 +108,176 @@ contract Agreement is IArbitrable, IForwarder, AragonApp {
 
         title = _title;
         arbitrator = _arbitrator;
-        collateralToken = _collateralToken;
+        stakingFactory = _stakingFactory;
 
-        _newSetting(_content, _delayPeriod, _settlementPeriod, _collateralAmount, _challengeCollateral);
-        _newTokenBalancePermission(_signPermissionToken, _signPermissionBalance, _challengePermissionToken, _challengePermissionBalance);
+        contents.length++; // Content zero is considered the null content for further validations
+        _newContent(_content);
     }
 
     /**
-    * @notice Stake `@tokenAmount(self.collateralToken(): address, _amount)` tokens for `_signer`
-    * @param _amount Number of collateral tokens to be staked by the sender
+    * @notice Sign the agreement
     */
-    function stake(uint256 _amount) external onlySigner(msg.sender) {
-        _stakeBalance(msg.sender, msg.sender, _amount);
-    }
+    function sign() external {
+        uint256 lastContentIdSigned = lastContentSignedBy[msg.sender];
+        uint256 currentContentId = _getCurrentContentId();
+        require(lastContentIdSigned < currentContentId, ERROR_SIGNER_ALREADY_SIGNED);
 
-    /**
-    * @notice Stake `@tokenAmount(self.collateralToken(): address, _amount)` tokens from `msg.sender` for `_signer`
-    * @param _signer Address staking the tokens for
-    * @param _amount Number of collateral tokens to be staked for the signer
-    */
-    function stakeFor(address _signer, uint256 _amount) external onlySigner(_signer) {
-        _stakeBalance(msg.sender, _signer, _amount);
+        lastContentSignedBy[msg.sender] = currentContentId;
+        emit Signed(msg.sender, currentContentId);
     }
 
     /**
-    * @dev Callback of `approveAndCall`, allows staking directly with a transaction to the token contract
-    * @param _from Address making the transfer
-    * @param _amount Amount of tokens to transfer
-    * @param _token Address of the token
+    * @notice Create a new action
+    * @param _disputableId Identification number of the disputable action in the context of the disputable instance
+    * @param _submitter Address of the user that has submitted the action
+    * @param _context Link to a human-readable text giving context for the given action
+    * @return Unique identification number for the created action in the context of the agreement
     */
-    function receiveApproval(address _from, uint256 _amount, address _token, bytes /* _data */) external onlySigner(_from) {
-        require(msg.sender == _token && _token == address(collateralToken), ERROR_SENDER_NOT_ALLOWED);
-        _stakeBalance(_from, _from, _amount);
-    }
+    function newAction(uint256 _disputableId, address _submitter, bytes _context) external returns (uint256) {
+        uint256 lastContentIdSigned = lastContentSignedBy[_submitter];
+        require(lastContentIdSigned >= _getCurrentContentId(), ERROR_SIGNER_MUST_SIGN);
```

Comment:
Not sure I follow, would it like signing a future content only? Actually, it should be only equal to but I added the greater than condition as well since it is not harmful, at least currently.

Reply:
Yes, that’s what I mean, that I don’t see how could be “greater than”, nor if it would make any sense. It would seem better to me to enforce `==`, easier to understand and even safer (like, if it’s greater something it’s wrong). But no big deal anyway.